gpslogger/src/main/java/com/mendhak/gpslogger/senders/sftp/SFTPManager.java

```diff
@@ -28,14 +31,13 @@ public void uploadFile(List<File> files) {
     }
 
     public void uploadFile(final File file){
-        final JobManager jobManager = AppSettings.getJobManager();
-        jobManager.cancelJobsInBackground(new CancelResult.AsyncCancelCallback() {
-            @Override
-            public void onCancelled(CancelResult cancelResult) {
-                jobManager.addJobInBackground(new SFTPJob(file, preferenceHelper.getSFTPRemoteServerPath(), preferenceHelper.getSFTPHost(),preferenceHelper.getSFTPPort(),preferenceHelper.getSFTPPrivateKeyFilePath(),
-                        preferenceHelper.getSFTPPrivateKeyPassphrase(),preferenceHelper.getSFTPUser(),preferenceHelper.getSFTPPassword(),preferenceHelper.getSFTPKnownHostKey()));
-            }
-        }, TagConstraint.ANY, SFTPJob.getJobTag(file));
+        String tag = String.valueOf(Objects.hashCode(file));
+        HashMap<String, Object> dataMap = new HashMap<String, Object>(){{
+            put("filePath", file.getAbsolutePath());
+        }};
+        OneTimeWorkRequest workRequest = Systems.getBasicOneTimeWorkRequest(SFTPWorker.class, dataMap);
+        WorkManager.getInstance(AppSettings.getInstance())
+                .enqueueUniqueWork(tag, androidx.work.ExistingWorkPolicy.REPLACE, workRequest);
```

Comment:
Could this entire bit be one call. 
Pass tag, dataMap, class to a method, and it does the enqueue.